app/src/main/java/com/zeapo/pwdstore/PasswordStore.kt

```diff
@@ -664,6 +666,52 @@ class PasswordStore : AppCompatActivity(R.layout.activity_pwdstore) {
         }.launch(intent)
     }
 
+    /**
+     * Prompt the user with a new category name to assign,
+     * if the new category forms/leads a path (i.e. contains "/"), intermediate directories will be created
+     * and new category will be placed inside.
+     *
+     * @param oldCategory The category to change its name
+     * @param error Determines whether to show an error to the user in the alert dialog, this error
+     * may be due to the new category the user entered already exists or the field was empty
+     */
+    private fun renameCategory(oldCategory: PasswordItem, error: Boolean = false) {
+        val view = layoutInflater.inflate(R.layout.folder_dialog_fragment, null)
+        val newCategoryEditText = view.findViewById<TextInputEditText>(R.id.folder_name_text)
+        if (error) {
+            newCategoryEditText.error = getString(R.string.message_error_rename_folder)
+        }
+
+        val dialog = MaterialAlertDialogBuilder(this)
+            .setTitle(R.string.title_rename_folder)
+            .setView(view)
+            .setMessage(getString(R.string.message_rename_folder, oldCategory.name))
+            .setPositiveButton(R.string.dialog_ok) { _, _ ->
+                val newCategory = File("${oldCategory.file.parent}/${newCategoryEditText.text}")
```

Comment:
We would probably also want to error out on path characters such as `/` and `\` in `newCategoryEditText.text`.

Reply:
With `/` , APS checks whether the new path exist or creates it with all needed folders before moving the category. But I don't know if `\` should generate an error, right now the folder can be named with `\` but I don't know if this could generate problems on windows compatibility due to their system separator is `\`.

Reply:
This is more of a general concern than one specific to your PR, but if users have their repository on a FAT-formatted SD card, we would definitely have to be wary of `\` and other special characters.

@msfjarvis Has this ever caused problems in practice? If so, that could be dealt with separately.

Reply:
Nope, never received a report about this.

Reply:
Well, then let's ignore it here as well.